velox/vector/fuzzer/VectorFuzzer.cpp

```diff
@@ -686,12 +731,29 @@ BufferPtr VectorFuzzer::fuzzNulls(vector_size_t size) {
 
 variant VectorFuzzer::randVariant(const TypePtr& arg) {
   VELOX_CHECK(arg->isPrimitiveType());
+  if (isDecimalKind(arg->kind())) {
+    return randomDecimalVariant(arg, rng_);
+  }
   return VELOX_DYNAMIC_SCALAR_TYPE_DISPATCH(
       randVariantImpl, arg->kind(), rng_, opts_);
 }
 
-TypePtr VectorFuzzer::randType(int maxDepth) {
-  static TypePtr kScalarTypes[]{
+std::pair<int8_t, int8_t> VectorFuzzer::randPrecisionScale(TypeKind kind) {
+  VELOX_DCHECK(isDecimalKind(kind));
+  // Generate precision in range [1, Decimal type max precision]
+  auto precision = 1 +
+      rand<int8_t>(rng_) %
+          (kind == TypeKind::SHORT_DECIMAL ? ShortDecimalType::kMaxPrecision
+                                           : LongDecimalType::kMaxPrecision);
+  // Generate scale in range [0, precision]
```

Comment:
typos?

Reply:
Sorry, not able to spot the typo here.
The scale will in the range 0 to precision inclusive.